projects/plugins/jetpack/_inc/client/components/navigation/index.jsx

```diff
@@ -43,13 +54,38 @@ export class Navigation extends React.Component {
 	};
 
 	trackRecommendationsClick = () => {
-		this.trackNavClick( 'recommendations' );
+		const isBubbleVisible = this.props.newRecommendationsCount > 0;
+
+		let recommendationsTrackData = {
+			target: 'nav_item',
+			path: 'recommendations',
+			is_new_recommendations_bubble_visible: isBubbleVisible,
+		};
+
+		if ( isBubbleVisible ) {
+			recommendationsTrackData = {
+				...recommendationsTrackData,
+				new_recommendations: this.props.newRecommendations ?? 'unknown',
+			};
+		}
+
+		analytics.tracks.recordJetpackClick( recommendationsTrackData );
 	};
 
 	trackMyJetpackClick = () => {
 		this.trackNavClick( 'my-jetpack' );
 	};
 
+	componentDidMount() {
+		this.trackNewRecommendations();
+	}
+
+	componentDidUpdate( prevProps ) {
```

Comment:
My thought process behind this was going back to handling all use cases no matter how unlikely. I know that `newRecommendationsCount` _shouldn't_ change unless the page is reloaded, but I felt like I should cover that case anyway. 

On the other side, I do understand that this may be completely unnecessary extra code that runs every time the page is updated 🤔 

I am not sure, after laying out my thought process what do you think @jboland88?

Reply:
Fair enough, I'll go ahead and remove it 😄